### PR TITLE
#119 pub date refactor

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #138 

refactored MarcPubDateProcessor to make it stateless and thread-safe
refactored process_publication_date as a macro and got rid of psulib_marc.rb
also refactored related specs

### CODE REVIEW

File: lib/traject/psulib_marc.rb [hunk 1]
```diff
@@ -1,27 +0,0 @@
-# frozen_string_literal: true
```
Comment (cdmo) [inline lib/traject/psulib_marc.rb]: Nice to delete this :) 🎉

File: lib/traject/psulib_config.rb [hunk 2]
```diff
@@ -208,10 +207,7 @@
 to_field 'publisher_manufacturer_tsim', extract_marc('260b:264|*1|b:260f:264|*3|b', trim_punctuation: true)
 
 ## Publication year facet (sidebar)
-to_field 'pub_date_sort_itsi' do |record, accumulator|
-  publication_date = process_publication_date record
-  accumulator << publication_date if publication_date
-end
+to_field 'pub_date_sort_itsi', process_publication_date
```
Comment (cdmo) [inline lib/traject/psulib_config.rb]: 🥇

Comment (banukutlu) [reply lib/traject/psulib_config.rb]: :)

File: lib/traject/psulib_marc.rb (hunk 1 above)
Comment (banukutlu) [reply lib/traject/psulib_marc.rb]: yes!

**Pull request merged**